react.js

```diff
@@ -16,15 +18,20 @@ export function wrap(fn) {
     stores[key] = writable(res[key]);
   }
 
+  effects.forEach(fn => fn());
+
   store.subscribe(function () {
     currentStore = store;
     i = 0;
+    effects = [];
 
     let res = fn();
     for (let key in res) {
       stores[key].set(res[key]);
     }
 
+    effects.forEach(fn => fn());
```

Comment:
I think the effects need to happen after the component updates? Do we need to wrap this in an `afterUpdate` block?

Also, I think useLayoutEffect and useEffect are slightly different in timing. useLayoutEffect is synchronous, and useEffect is asynchronous. Maybe need a `requestAnimationFrame` or something in there for the async one?

Reply:
good idea! updated the code.

Reply:
added 2 different effect queeus.